src/plugman/fetch.js

```diff
@@ -63,22 +63,10 @@ function fetchPlugin (plugin_src, plugins_dir, options) {
         if (result) {
             if (result[1]) { options.git_ref = result[1]; }
             if (result[2]) { options.subdir = result[2]; }
-            // if --fetch was used, throw error for subdirectories
 
+            // throw error for subdirectories
             if (options.subdir && options.subdir !== '.') {
-                events.emit('warn', 'support for subdirectories is deprecated and will be removed in Cordova@7');
-                if (options.fetch) {
-                    return Q.reject(new CordovaError('--fetch does not support subdirectories'));
-                }
-            }
-
-            // Recurse and exit with the new options and truncated URL.
-            var new_dir = plugin_src.substring(0, plugin_src.indexOf('#'));
-
-            // skip the return if user asked for --fetch
-            // cordova-fetch doesn't need to strip out git-ref
-            if (!options.fetch) {
-                return fetchPlugin(new_dir, plugins_dir, options);
+                return Q.reject(new CordovaError('Cordova does not support subdirectories'));
```

Comment:
Ugh, this broke the documented way to install plugin tests: https://github.com/apache/cordova-plugin-test-framework#tldr-try-it

Reply:
This is also still documented at https://github.com/apache/cordova-cli/blob/master/doc/readme.md#plugin-spec

Reply:
That `:subdirectory` syntax was specific to the old Cordova fetching code (which has been removed), and is not supported by npm

Reply:
https://github.com/apache/cordova-lib/issues/722